src/containers/load-button.jsx

```diff
@@ -13,10 +18,23 @@ class LoadButton extends React.Component {
             'handleChange',
             'handleClick'
         ]);
+        this.state = {
+            loadingError: false,
+            errorMessage: ''
+        };
     }
     handleChange (e) {
+        this.props.openLoadingState();
+        // Remove the hash if any (without triggering a hash change event or a reload)
+        history.replaceState({}, document.title, '.');
```

Comment:
Things that modify the state of the page should not be buried in individual components, and should instead be handled by redux or a HOC that the "GUI component user" has control over.  I think our best bet atm would be to get this behavior into the `ProjectLoaderHOC` somehow, since that already is handling hash changes.

Reply:
@rschamp, I reverted the changes we were working on (as a fix for this). Could you take another look?